semgrep_output_v1.atd

```diff
@@ -868,6 +868,124 @@ type cli_output_extra = {
     ~skipped_rules: skipped_rule list;
 }
 
+(*****************************************************************************)
+(* Matching diagnosis *)
+(*****************************************************************************)
+
+(* A "matching diagnosis" is a postprocessed interpretation of matching
+   explanations, specific to a particular test-annotated target file.
+
+   For instance, suppose we have the rule:
+   1 | all:
+   2 | - pattern: foo(...)
+   3 | - not: foo(good)
+   and the following Python annotated target:
+
+   1 | # ruleid: my_rule
+   2 | foo()
+   3 | # ok: my_rule
+   4 | foo(bad)
```

Comment:
maybe use something like foo(good),  instead of foo(bad).

Reply:
So it's clear that the problem is in the rule, which should be not: foo(good) instead of not: foo(goood).
after all that's the goal of matching diagnosis, to help fix the rule and diagnose.